pkgs/by-name/wi/wivrn/package.nix

```diff
@@ -0,0 +1,130 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchFromGitLab
+, avahi
+, boost
+, cmake
+, cudaPackages
+, eigen
+, ffmpeg
+, freetype
+, git
+, glm
+, glslang
+, harfbuzz
+, libdrm
+, libva
+, libpulseaudio
+, libX11
+, libXrandr
+, monado
+, nlohmann_json
+, onnxruntime
+, openxr-loader
+, pkg-config
+, python3
+, shaderc
+, spdlog
+, systemd
+, udev
+, vulkan-headers
+, vulkan-loader
+, vulkan-tools
+, x264
+}:
+let
+  vendorMonado = monado.overrideAttrs rec {
+    # Version stated in CMakeList for WiVRn 0.11
+    version = "57e937383967c7e7b38b5de71297c8f537a2489d";
+
+    src = fetchFromGitLab {
+      domain = "gitlab.freedesktop.org";
+      owner = "monado";
+      repo = "monado";
+      rev = version;
+      hash = "sha256-O/Td2WccTr4Fa8U64/lVfnidSIH5t3gWuFXCsEVf7bk=";
+    };
+
+    postInstall = ''
+      mv src/xrt/compositor/libcomp_main.a $out/lib/libcomp_main.a
+    '';
```

Comment:
I'm having issues with even getting xrgears to work now for some reason so I can't even test if this is needed. It was there in the other package so I figured it was needed.

Reply:
@PassiveLemon (just in case that didn't come through) note that `postInstall` is defined in a let-in expression, but never consumed in the `mkDerivation`

Reply:
Also I don't know what `libcomp_main.a` is but if it needs to be installed for some reason then it's better to submit a patch updating CMakeLists.txt to do the installation

Reply:
It's probably in the monado package's jurisdiction, but I don't see this line there.

Reply:
As we only use `vendorMonado.src` below, we should be able to get rid of this completely and only keep the pinned Monado source (the value of `fetchFromGitLab`) in a variable

Reply:
The following diff builds fine for me:
```diff
diff --git a/pkgs/by-name/wi/wivrn/package.nix b/pkgs/by-name/wi/wivrn/package.nix
index ab5903e69059..a837ece28560 100644
--- a/pkgs/by-name/wi/wivrn/package.nix
+++ b/pkgs/by-name/wi/wivrn/package.nix
@@ -18,7 +18,6 @@
 , libpulseaudio
 , libX11
 , libXrandr
-, monado
 , nlohmann_json
 , onnxruntime
 , openxr-loader
@@ -33,35 +32,25 @@
 , vulkan-tools
 , x264
 }:
-let
-  vendorMonado = monado.overrideAttrs rec {
-    # Version stated in CMakeList for WiVRn 0.11
-    version = "57e937383967c7e7b38b5de71297c8f537a2489d";
-
-    src = fetchFromGitLab {
-      domain = "gitlab.freedesktop.org";
-      owner = "monado";
-      repo = "monado";
-      rev = version;
-      hash = "sha256-O/Td2WccTr4Fa8U64/lVfnidSIH5t3gWuFXCsEVf7bk=";
-    };
-
-    postInstall = ''
-      mv src/xrt/compositor/libcomp_main.a $out/lib/libcomp_main.a
-    '';
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "wivrn";
   version = "0.11";
 
   src = fetchFromGitHub {
     owner = "meumeu";
     repo = "wivrn";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     hash = "sha256-E48JYhkBVZfb7S/FW0F8RbMtx4GIJwfXfs4KAF3gn8A=";
   };
 
+  monadoSrc = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "monado";
+    repo = "monado";
+    rev = "57e937383967c7e7b38b5de71297c8f537a2489d";
+    hash = "sha256-O/Td2WccTr4Fa8U64/lVfnidSIH5t3gWuFXCsEVf7bk=";
+  };
+
   # The library path to the OpenXR runtime requires a relative path from the config file to the binary in the nix store
   # The config file is usually located at ~/.config/openxr/1/ but the wivrn module puts it at /etc/xdg/openxr/1/
   # The CMakeList has relative directory paths that cause malformation of the path. https://github.com/Meumeu/WiVRn/issues/47
@@ -115,7 +104,7 @@ stdenv.mkDerivation rec {
     (lib.cmakeBool "WIVRN_USE_X264" true)
     (lib.cmakeBool "WIVRN_USE_NVENC" false)
     (lib.cmakeBool "FETCHCONTENT_FULLY_DISCONNECTED" true)
-    (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_MONADO" "${vendorMonado.src}")
+    (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_MONADO" "${finalAttrs.monadoSrc}")
   ];
 
   meta = with lib; {
@@ -127,4 +116,4 @@ stdenv.mkDerivation rec {
     platforms = platforms.linux;
     mainProgram = "wivrn-server";
   };
-}
+})
```

Reply:
This diff is great.